utils/rpms/daos.spec

```diff
@@ -562,6 +559,7 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 %{_libdir}/libgurt.so
 %{_libdir}/libcart.so
 %{_libdir}/*.a
+%{daoshome}/python
```

Comment:
What's in this dir that is `-devel` specific?

Reply:
The pydaos sources.  We could put them elsewhere but they need the daos headers to build/install so they need daos-devel installed in order to be useful.

Reply:
I guess the problem here is my ignorance of pydaos.  Who would use the sources?  End-users?  Python developers that wanted to develop python packages for use with DAOS?

Do the pydaos sources not build into, say, a library that would be pydaos-user usable?

How was anyone getting pydaos before this change?

Reply:
pydaos in a python module for accessing daos from python.  It's installed as part of "scons install" but you can also pip install it, see src/client/setup.py and src/client/pydaos.

Our daos-client rpm includes a built version of it but only for python 3.6 and it's installed into "site-packages" so it's not available to virtual environments so in practice it's pretty much unusable now.

Initially in this PR I included the pydaos sources into the daos-client rpm so that they could be pip installed for other python versions etc but as it's a C based daos client it needs the daos headers and hence the daos-devel rpm.  The details are included in https://daosio.atlassian.net/browse/DAOS-14849 this PR fixes two tickets.

Reply:
`daos-devel` just doesn't seem right for this, but other than building it as a proper python module (which does not solve the _venv_ problem I'm not sure what to suggest.